controllers/idler/idler_controller.go

```diff
@@ -111,6 +114,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(ctx, idler, r.setStatusFailed, err,
 			"failed to ensure idling '%s'", idler.Name)
 	}
+	if err := r.ensureAnsiblePlatformIdling(ctx, idler); err != nil {
```

Comment:
shouldn't we move this before the `ensureIdling` ? 

In this way we first check if there's any pod associated with AAP and only after that we idle remaining pods.

Reply:
I was thinking about it but too but it doesn't really matter, does it? We can't rely on the order inside the reconcile loop anyway. Even if the ensureAAPIdling is executed first and does idle the AAP CRs, the pods actually would be deleted with a noticeable delay, so ensureIdling will be executed and finished before the pods are deleted by the AAP operator. So it wouldn't solve the problem with two idler functions (the aap and the main one) trying to idle the same aap pods. Instead it relies on the shorten AAP idling timeout. See https://github.com/codeready-toolchain/member-operator/pull/631#discussion_r1980090798 for details.

Reply:
Correct the logic shouldn't rely on the order we execute the functions, but still I was thinking it might help with processing less pods in `ensureIdling`. The pods won't be deleted immediately right, but there are chances some of them might be gone by the time we start the `ensureIdling` , or at least they should be in `Terminating` state thus the function should ignore them. 

So basically this would be just a performance improvement.

Reply:
My experience tells me that it takes at least seconds to get the pod in the terminating state. So it doesn't look like there is a good chance that the pods will be gone by the time of the ensureIdling execution. Also there is actually another reason for keeping ensureIdling first. We still need the pods to be tracked in the Idler.Status which is done ensureIdling.

Reply:
ok makes sense!

 Out of curiosity why do we need to pods to be tracked in Idler.Status ?

Reply:
We don't have to. It was the original design (so we just have something in the status which could help us with debugging problems with the idler) but now it's used in the main idler. We can get rid of it but it would requires some refactoring.

Reply:
>  We can get rid of it but it would requires some refactoring.

I was thinking about the same, that the list of pods is actually not needed and could be easily removed. As for the refactoring, it would be really small and it would actually simplify the logic a bit. 

I've created a story for it https://issues.redhat.com/browse/KUBESAW-269